### PR TITLE
fix: crash in `bun create -h` #7971

### PR DESCRIPTION
**Pull request by coratgerl**
### What does this PR do?

Fixes: #7971 
Refactor the create command to match with the architecture of the other command.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

I wrote automated tests 

- [x] I checked the lifetime of memory allocated to verify it's (1) freed and (2) only freed when it should be
- [x] I included a test for the new code, or an existing test covers it
- [x] JSValue used outside outside of the stack is either wrapped in a JSC.Strong or is JSValueProtect'ed
- [x] I wrote TypeScript/JavaScript tests and they pass locally (`bun-debug test test-file-name.test`)

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
It should print the help menu, not a new error

**Comment by coratgerl** [comment]
Indeed, I made the change to display the help menu.

- For this I made a refactor to have the same architecture as for the other command (the file is more clear now)
- I also fixed a test that I don't understand how it passed before, indeed, on the screen below we can see the template is create under the `bun-create` folder(in main branch), this is the expected behavior but the test checked if file exist in `test-template`and not in bun-create as you can see on the second screen :  
```javascript 
const bunCreateDir = join(x_dir, "bun-create");
  const testTemplate = "test-template";

  await mkdir(`${bunCreateDir}/${testTemplate}`, { recursive: true });
  const { exited } = spawn({
    cmd: [bunExe(), "create", testTemplate],
    stdout: null,
    stdin: "pipe",
    stderr: "pipe",
    cwd: x_dir,
    env: { ...env, BUN_CREATE_DIR: bunCreateDir },
  });

  await exited;

  const dirStat = await stat(`${x_duir}/${testTemplate}`);
  expect(dirStat.isDirectory()).toBe(true);
```

 
image

File: src/cli.zig [hunk 1]
```diff
@@ -1988,27 +2022,31 @@ pub const Command = struct {
...
+                        \\<b>Usage<r>: <b><green>bun create<r> <cyan>\<template\><r> [\<destination\>]
+                        \\<b>Alias: <b>bun c<r>
+                        \\Create a new bun project using a create-\<template\> npm package, GitHub repo, or a local template
                     ;
 
                     const outro_text =
-                        \\If given a GitHub repository name, Bun will download it and use it as a template,
-                        \\otherwise it will run <b><magenta>bunx create-\<template\><r> with the given arguments.
+                        \\<b>Examples:<r>
+                        \\  <b>From npm :<r>
```
Comment (paperclover) [inline src/cli.zig]:
```suggestion
                        \\  <d>From npm<r>
```

File: src/cli.zig [hunk 2]
```diff
@@ -1988,27 +2022,31 @@ pub const Command = struct {
...
+                        \\  <b><green>bun create remix<r>
+                        \\
+                        \\  <b>From GitHub :<r>
+                        \\  <b><green>bun create github.com/\<user\>/\<repo\><r>
```
Comment (paperclover) [inline src/cli.zig]:
```suggestion
                        \\  <d>From GitHub<r>
                        \\  <b><green>bun create \<user\>/\<repo\><r>
```
maybe we can pick an actual example, idk

**Review by paperclover — CHANGES_REQUESTED** [review]

File: src/cli.zig (hunk 2 above)
Comment (coratgerl) [reply src/cli.zig]: I think we can keep the basic example of the doc as you put in your suggestion. The link to the full documentation is in the bottom of the man command :)

File: src/cli.zig [hunk 3]
```diff
@@ -1988,27 +2022,31 @@ pub const Command = struct {
...
+                        \\<b>Examples:<r>
+                        \\  <d>From npm<r>
+                        \\  <b><green>bun create remix<r>
+                        \\
+                        \\  <d>From GitHub<r>
+                        \\  <b><green>bun create \<user\>/\<repo\><r>
                         \\
-                        \\Learn more about creating new projects: <magenta>https://bun.sh/docs/cli/bun-create<r>
+                        \\  <d>From local template<r>
+                        \\  <b><green>bun create $HOME/.bun-create/\<name\><r>
```
Comment (paperclover) [inline src/cli.zig:2088]: wait does this work?

Comment (coratgerl) [reply src/cli.zig:2088]: I took this elements from the documentation but I don't tested. If the documentation is correct I suppose yes it's work

**Review by paperclover — APPROVED** [review]
sorry for late review. this only really needed to change one thing but the rest seems ok.

**Comment by coratgerl** [comment]
Thanks for your feedback, the conflict is fixed @paperdave

**Pull request closed without merging**